src/api/nodeRpcCall.js

```diff
@@ -1924,6 +1924,63 @@ const fetchPendingIdentities = async () => {
   return processed.filter((entity) => entity.data.judgements.length === 0);
 };
 
+const getSignaturesForGivenId = async (allSignatures, id) => allSignatures.find(
+  ([contractId]) => contractId.args[0].eq(id),
+)[1].unwrapOrDefault();
+
+const getAllContracts = async () => {
+  const api = await getApi();
+
+  const [rawJudgesSignatures, rawPartiesSignatures] = await Promise.all([
+    api.query.contractsRegistry.judgesSignatures.entries(),
+    api.query.contractsRegistry.partiesSignatures.entries(),
+  ]);
+
+  const rawContracts = await api.query.contractsRegistry.contracts.entries();
+  return Promise.all(rawContracts.map(async ([id, contract]) => {
+    const contractId = id.args[0];
+    const [judgesSignatures, partiesSignatures] = await Promise.all([
+      getSignaturesForGivenId(rawJudgesSignatures, contractId),
+      getSignaturesForGivenId(rawPartiesSignatures, contractId),
+    ]);
+
+    const shouldBeData = contract.unwrapOr(null);
+    return {
+      contractId,
+      data: shouldBeData ? shouldBeData.data : null,
```

Comment:
After doing the `shouldBeData` -> `contract` rename from previous comment you can use this instead of ternary operator:

```suggestion
      data: contract?.data,
```

And same for parties/creator/deposit.